README.md
#Chinese Practice


history.txt
召,zhao,4,pass,Fri Aug 14 23:43:27 2020
胞,bao,1,pass,Fri Aug 14 23:43:27 2020
肌,ji,1,pass,Fri Aug 14 23:43:27 2020
鲸,jing,1,pass,Fri Aug 14 23:43:27 2020
褐,he,4,pass,Fri Aug 14 23:43:27 2020
煤,mei,2,pass,Fri Aug 14 23:43:27 2020
鼾,han,1,pass,Fri Aug 14 23:43:27 2020
窃,qie,4,pass,Fri Aug 14 23:43:27 2020
贼,zei,2,pass,Fri Aug 14 23:43:27 2020
秦,zou,4,pass,Fri Aug 14 23:43:27 2020
旋,xuan,2,pass,Fri Aug 14 23:43:27 2020
援,huan,3,fail,Fri Aug 14 23:43:27 2020
脊,ji,3,pass,Fri Aug 14 23:43:27 2020
测,ce,4,pass,Fri Aug 14 23:43:27 2020
攒,zan,3,pass,Fri Aug 14 23:43:27 2020
贩,fan,4,pass,Fri Aug 14 23:43:27 2020
墓,mu,4,pass,Fri Aug 14 23:43:27 2020
坟,fen,2,pass,Fri Aug 14 23:43:27 2020
迁,qian,2,fail,Fri Aug 14 23:43:27 2020
葬,?,4,fail,Fri Aug 14 23:43:27 2020
茄,qie,2,pass,Fri Aug 14 23:43:27 2020
赠,zeng,4,pass,Fri Aug 14 23:43:27 2020
捐,juan,1,pass,Fri Aug 14 23:43:27 2020
裕,yu,4,pass,Fri Aug 14 23:43:27 2020
隐,wen,1,fail,Fri Aug 14 23:43:27 2020
供,gong,1,pass,Fri Aug 14 23:43:27 2020
杆,gan,1,pass,Fri Aug 14 23:43:27 2020
谋,mou,2,pass,Fri Aug 14 23:43:27 2020
绣,xiu,4,pass,Fri Aug 14 23:43:27 2020
征,zheng,1,pass,Fri Aug 14 23:43:27 2020
暇,xia,2,pass,Fri Aug 14 23:43:27 2020
衬,fu,2,fail,Fri Aug 14 23:43:27 2020
妆,zhuang,1,pass,Fri Aug 14 23:43:27 2020
荤,hun,1,pass,Fri Aug 14 23:43:27 2020
牲,sheng,1,pass,Fri Aug 14 23:57:07 2020
畜,chu,4,pass,Fri Aug 14 23:57:07 2020
禽,qin,2,pass,Fri Aug 14 23:57:07 2020
奏,zou,4,pass,Fri Aug 14 23:57:07 2020
锣,luo,2,pass,Fri Aug 14 23:57:07 2020
袍,bao,1,fail,Fri Aug 14 23:57:07 2020
屁,lu,2,fail,Fri Aug 14 23:57:07 2020
录,lu,1,fail,Fri Aug 14 23:57:07 2020
驱,qu,1,pass,Fri Aug 14 23:57:07 2020
尘,chen,2,pass,Fri Aug 14 23:57:07 2020
鬼,gui,3,pass,Fri Aug 14 23:57:07 2020
遗,yi,2,pass,Fri Aug 14 23:57:07 2020
祭,ji,4,pass,Fri Aug 14 23:57:07 2020
触,chu,4,pass,Fri Aug 14 23:57:07 2020
怨,yuan,4,pass,Fri Aug 14 23:57:07 2020
幼,you,4,pass,Fri Aug 14 23:57:07 2020
携,xie,2,pass,Fri Aug 14 23:57:07 2020
纱,miao,3,fail,Fri Aug 14 23:57:07 2020
潮,chao,2,pass,Fri Aug 14 23:57:07 2020
燃,ran,2,pass,Fri Aug 14 23:57:07 2020
炮,pao,4,pass,Fri Aug 14 23:57:07 2020
噼,pi,1,pass,Fri Aug 14 23:57:07 2020
矩,ju,4,fail,Fri Aug 14 23:57:07 2020
掺,chan,1,pass,Fri Aug 14 23:57:07 2020
逛,guang,4,pass,Fri Aug 14 23:57:07 2020
醋,cu,4,fail,Fri Aug 14 23:57:07 2020
蜡,la,4,pass,Fri Aug 14 23:57:07 2020
烛,zhu,2,pass,Fri Aug 14 23:57:07 2020
朴,pu,4,fail,Sat Aug 15 00:59:25 2020
喉,hou,2,pass,Sat Aug 15 00:59:25 2020
稻,dao,4,pass,Sat Aug 15 00:59:25 2020
缰,jiang,1,pass,Sat Aug 15 00:59:25 2020
恳,ken,3,pass,Sat Aug 15 00:59:25 2020
逊,sui,2,fail,Sat Aug 15 00:59:25 2020
恰,qia,4,pass,Sat Aug 15 00:59:25 2020
宠,chong,4,fail,Sat Aug 15 00:59:25 2020
堪,kan,1,pass,Sat Aug 15 00:59:25 2020
款,kuan,3,fail,Sat Aug 15 00:59:25 2020
僵,jiang,1,pass,Sat Aug 15 00:59:25 2020
沾,zhan,1,pass,Sat Aug 15 00:59:25 2020
雅,ya,3,pass,Sat Aug 15 00:59:25 2020
叭,ba,1,pass,Sat Aug 15 00:59:25 2020
唧,ji,1,pass,Sat Aug 15 00:59:25 2020
膊,bo,2,pass,Sat Aug 15 00:59:25 2020
胳,ge,1,pass,Sat Aug 15 00:59:25 2020
授,shou,4,pass,Sat Aug 15 00:59:25 2020
例,li,4,pass,Sat Aug 15 00:59:25 2020
赴,fu,4,pass,Sat Aug 15 00:59:25 2020
基,ji,1,pass,Sat Aug 15 00:59:25 2020
邀,yao,1,pass,Sat Aug 15 00:59:25 2020
惯,guan,4,pass,Sat Aug 15 00:59:25 2020
壶,hu,2,pass,Sat Aug 15 00:59:25 2020
辈,bei,4,pass,Sat Aug 15 00:59:25 2020
乙,yi,4,fail,Sat Aug 15 00:59:25 2020
珊,yi,4,fail,Sat Aug 15 00:59:25 2020
零,yi,4,fail,Sat Aug 15 00:59:25 2020
堡,bao,3,pass,Sat Aug 15 00:59:25 2020
悦,yue,4,pass,Sat Aug 15 00:59:25 2020
载,can,2,fail,Sat Aug 15 00:59:25 2020
嘱,zhu,3,pass,Sat Aug 15 00:59:25 2020
侍,shou,4,fail,Sat Aug 15 00:59:25 2020
乞,yi,3,fail,Sat Aug 15 00:59:25 2020
纠,jiu,1,pass,Sat Aug 15 00:59:25 2020
柴,chai,2,pass,Sat Aug 15 00:59:25 2020
赤,chi,4,pass,Sat Aug 15 00:59:25 2020
涛,tao,1,pass,Sat Aug 15 00:59:25 2020
某,mei,2,fail,Sat Aug 15 00:59:25 2020
质,zhi,4,pass,Sat Aug 15 00:59:25 2020
吁,hu,1,fail,Sat Aug 15 00:59:25 2020
叽,hu,1,fail,Sat Aug 15 00:59:25 2020
喳,zha,1,pass,Sat Aug 15 00:59:25 2020
莫,mou,2,fail,Sat Aug 15 00:59:25 2020
覆,xia,2,fail,Sat Aug 15 00:59:25 2020
熬,ao,2,pass,Sat Aug 15 00:59:25 2020
昆,kun,1,pass,Sat Aug 15 00:59:25 2020
杜,zhuang,1,fail,Sat Aug 15 00:59:25 2020
翠,bi,4,fail,Sat Aug 15 00:59:25 2020
辞,ci,2,pass,Sat Aug 15 00:59:25 2020
播,bo,1,pass,Sat Aug 15 00:59:25 2020
迫,po,4,pass,Sat Aug 15 00:59:25 2020
启,qi,3,pass,Sat Aug 15 00:59:25 2020
傅,fu,4,pass,Sat Aug 15 00:59:25 2020
爆,bao,4,pass,Sat Aug 15 00:59:25 2020
酱,jiang,4,pass,Sat Aug 15 00:59:25 2020
彻,qie,4,fail,Sat Aug 15 00:59:25 2020
掀,si,1,fail,Sat Aug 15 00:59:25 2020
茶,cha,2,pass,Sat Aug 15 00:59:25 2020
舍,she,3,pass,Sat Aug 15 00:59:25 2020
席,xi,2,pass,Sat Aug 15 00:59:25 2020
仁,ren,2,pass,Sat Aug 15 00:59:25 2020
倾,qing,1,pass,Sat Aug 15 00:59:25 2020
均,jun,1,pass,Sat Aug 15 00:59:25 2020
匀,xi,2,fail,Sat Aug 15 00:59:25 2020
舅,jiu,4,pass,Sat Aug 15 00:59:25 2020
铺,pu,1,pass,Sat Aug 15 00:59:25 2020
俱,ju,4,pass,Sat Aug 15 00:59:25 2020
煎,jian,1,pass,Sat Aug 15 00:59:25 2020
描,miao,2,pass,Sat Aug 15 00:59:25 2020
烹,peng,1,pass,Sat Aug 15 00:59:25 2020
焰,bao,4,fail,Sat Aug 15 00:59:25 2020
愈,yu,4,pass,Sat Aug 15 00:59:25 2020
废,fa,1,fail,Sat Aug 15 00:59:25 2020
壳,ke,2,pass,Sat Aug 15 00:59:25 2020
腐,fu,3,pass,Sat Aug 15 00:59:25 2020
蒸,peng,1,fail,Sat Aug 15 00:59:25 2020
饮,yin,3,pass,Sat Aug 15 00:59:25 2020
雁,yan,4,pass,Sat Aug 15 00:59:25 2020
拇,mu,3,pass,Sat Aug 15 00:59:25 2020
瑜,qiu,3,fail,Sat Aug 15 00:59:25 2020
呵,he,1,pass,Sat Aug 15 00:59:25 2020
占,zhan4,1,fail,Sat Aug 15 00:59:25 2020
撤,zhan4,1,fail,Sat Aug 15 00:59:25 2020
签,jian,3,fail,Sat Aug 15 00:59:25 2020
魏,gui,3,fail,Sat Aug 15 00:59:25 2020
致,zhi,4,pass,Sat Aug 15 00:59:25 2020
琴,qin,2,pass,Sat Aug 15 00:59:25 2020
归,gui,1,pass,Sat Aug 15 00:59:25 2020
宴,wei,3,fail,Sat Aug 15 00:59:25 2020
踏,ta,4,pass,Sat Aug 15 00:59:25 2020

main copy.py
from tkinter import*
import random
import datetime
import math
import configparser
import os
from time import gmtime, strftime

rank_of_word = 0
list_of_objects = []
list_of_answers = []
words = []

ScriptPath = os.path.abspath(__file__)
#a = ScriptPath.index('main.py')
#a = ScriptPath[:-7]
#b = 'main.ini'
#IniPath = a+b

#config = configparser.ConfigParser()
##config.read(IniPath, encoding='utf-8')
#FilePaths = config['data']
DictionaryFilePath = 'C:\\Users\\alexl.HOME02\\OneDrive\\ChineseWords\\data\\dictionary.txt'
HistoryFilePath = 'C:\\Users\\alexl.HOME02\\OneDrive\\ChineseWords\\data\\history.txt'

class TestRecord:
    def __init__(self):
        #self.round = 0
        #self.index = 0
        self.character = ''
        self.pinyin = ''
        self.yindiao = 1
        self.passed = 'fail'
        #self.seconds_spent = 0
        self.time = strftime("%c")


    def add_to_history(self, filepath):
        document = open(filepath, 'a', encoding='utf-8')
        character = str(self.character)
        pinyin = str(self.pinyin)
        yindiao = str(self.yindiao)
        right_or_wrong = str(self.passed)
        time =  str(self.time)
        document.write('\n')
        document.write(character)
        document.write(',')
        document.write(pinyin)
        document.write(',')
        document.write(yindiao)
        document.write(',')
        document.write(right_or_wrong)
        document.write(',')
        document.write(time)

    def check_if_coorect(self, pinyin_answer, yindiao_answer):
        if pinyin_entry.get() == pinyin_answer and yindiao_entry.get() == yindiao_answer:
            self.passed = 'pass'
            return True
        
        else:
            self.passed = 'fail'
            return False

class ChineseAnswers:

    def __init__(self):
        self.character = ''
        self.pinyin = ''
        self.yindiao = 0


def read_dictionary():
    global list_of_answers
    doc = open(DictionaryFilePath, 'r', encoding='utf-8')
    tempstr = doc.read()
    a = tempstr.split('\n')
    list_of_answers = []
    for c in a:
        b = c.split(',')

        ca = ChineseAnswers()
        ca.character = b[0]
        ca.pinyin = b[1]
        ca.yindiao = int(b[2])
        list_of_answers.append(ca)
    return list_of_answers

def read_history():
    global list_of_objects
    doc = open(HistoryFilePath, 'r', encoding='utf-8')
    tempstr = doc.read()
    a = tempstr.split('\n')
    list_of_objects = []
    for c in a:
        b = c.split(',')

        #__init__ is used
        tr = TestRecord()
        tr.character = b[0]
        tr.pinyin = b[1]
        tr.yindiao = int(b[2])
        tr.passed = b[3]
        list_of_objects.append(tr)

    return list_of_objects

def groupby(list):
    result = {}

    for i in range(0, len(list)):
        current_object = list[i]
        key = current_object.character

        if key in result:
            temp_list = result.get(key)
            temp_list.append(current_object)
            result[key] = temp_list
        else:
            result[key] = [current_object]

    return result

    

def sort_percentages(dictionary):
    global words
    number_of_trys = 0
    percentage_dictionary = {}
    list_of_objects = []
    for key in dictionary:
        number_fail = 0
        number_pass = 0
        list_of_objects = dictionary.get(key)

        for j in range(0, len(list_of_objects)):
            if list_of_objects[j].passed == True:
                number_pass = number_pass + 1
                number_of_trys = number_of_trys + 1
            else:
                number_fail = number_fail + 1
                number_of_trys = number_of_trys + 1

        percentage = float((number_pass - number_fail))
        percentage_dictionary[key] = percentage
    
    percentage_list = sorted(percentage_dictionary.items(), key=lambda x: x[1])
    percentage_list.reverse()

    return percentage_list
    
a = read_dictionary()
dictionary_of_answers = groupby(read_dictionary())
dictionary = groupby(read_history())
percentage_list = sort_percentages(dictionary)

def next_word():
    Chinese_Word_Variable2.set('')
    Chinese_Word_Variable3.set('')
    global dictionary_of_answers
    global dictionary
    global percentage_list
    global rank_of_word

    current_word = TestRecord()
    #entered word and pinyin and yindiaos
    current_word.character = percentage_list[rank_of_word][0]
    current_word.pinyin = pinyin_entry.get()
    current_word.yindiao = yindiao_entry.get()
    
    Chinese_Word_Variable2.set('')
    Chinese_Word_Variable3.set('')

    #coorect pinyin and yindiao
    coorect_word = dictionary_of_answers.get(current_word.character)
    Chinese_Word_Variable.set(current_word.character)
    pinyin_answer = coorect_word[0].pinyin
    yindiao_answer = coorect_word[0].yindiao

    current_word.pinyin = pinyin_entry.get()
    current_word.yindiao = yindiao_entry.get()
    if yindiao_entry.get() != '' and current_word.character != '':

        

        if current_word.check_if_coorect(pinyin_answer, str(yindiao_answer)):
            current_word.add_to_history('C:\\Users\\alexl.HOME02\\OneDrive\\ChineseWords\\data\\history.txt')
            rank_of_word = rank_of_word + 1
            current_word = TestRecord()
            Chinese_Word_Variable2.set('')
            Chinese_Word_Variable3.set('')
            Chinese_Word_Variable.set(current_word.character)

        else:
            current_word.add_to_history('C:\\Users\\alexl.HOME02\\OneDrive\\ChineseWords\\data\\history.txt')
            pinyin_answer = coorect_word[0].pinyin
            yindiao_answer = coorect_word[0].yindiao
            Chinese_Word_Variable2.set(str(pinyin_answer))
            Chinese_Word_Variable3.set(str(yindiao_answer))


app = Tk()
app.title('Chinese Pratice')
app.geometry('1000x640')
btn = Button(app, text='next word', command=next_word, height=12, padx = 470).grid(row=0, sticky=W)
Chinese_Word_Variable = StringVar()
Chinese_Word_Variable2 = StringVar()
Chinese_Word_Variable3 = StringVar()
Chinese_Word_Label = Label(app, textvariable=Chinese_Word_Variable, padx = 20).grid(row=1, column=0)
Chinese_Word_Label2 = Label(app, textvariable=Chinese_Word_Variable2, padx = 20).grid(row=2)
Chinese_Word_Label3 = Label(app, textvariable=Chinese_Word_Variable3, padx = 20).grid(row=3)
#Chinese_Word_Variable.set(current_word.character)

#make entrys for pinyin and yindiao
pinyin_label = Label(app, text='enter pinyin').grid(row=4, sticky=W)
yindiao_label = Label(app, text='enter yindiao').grid(row=5, sticky=W)
pinyin_entry = Entry(app)
pinyin_entry.grid(row=4)
yindiao_entry = Entry(app)
yindiao_entry.grid(row=5)
entered_yindiao = 0
entered_pinyin = 0

app.mainloop()

main.py
from tkinter import*
import random
import datetime
import math
import configparser
import os
from time import gmtime, strftime

rank_of_word = 0
list_of_objects = []
list_of_answers = []
words = []

date = datetime.datetime.now()

count = 1

ScriptPath = os.path.abspath(__file__)
#a = ScriptPath.index('main.py')
#a = ScriptPath[:-7]
#b = 'main.ini'
#IniPath = a+b

#config = configparser.ConfigParser()
##config.read(IniPath, encoding='utf-8')
#FilePaths = config['data']
DictionaryFilePath = 'C:\\Users\\alexl.HOME02\\OneDrive\\ChineseWords\\data\\dictionary.txt'
HistoryFilePath = 'C:\\Users\\alexl.HOME02\\OneDrive\\ChineseWords\\data\\history.txt'

class TestRecord:
    def __init__(self):
        #self.round = 0
        #self.index = 0
        self.character = ''
        self.pinyin = ''
        self.yindiao = 1
        self.passed = 'fail'
        #self.seconds_spent = 0
        self.time = date.strftime('%c')


    def add_to_history(self, filepath):
        date = datetime.datetime.now()
        document = open(filepath, 'a', encoding='utf-8')
        character = str(self.character)
        pinyin = str(self.pinyin)
        yindiao = str(self.yindiao)
        right_or_wrong = str(self.passed)
        time =  date.strftime('%c')
        document.write('\n')
        document.write(character)
        document.write(',')
        document.write(pinyin)
        document.write(',')
        document.write(yindiao)
        document.write(',')
        document.write(right_or_wrong)
        document.write(',')
        document.write(date.strftime('%c'))

    def check_if_coorect(self, pinyin_answer, yindiao_answer):
        pinyin1 = pinyin_answer[0]
        pinyin2 = '    '
        pinyinentry = pinyin_entry.get()
        if len(pinyin_answer) == 2:  
            pinyin2 = pinyin_answer[1]

        if pinyin_entry.get() in pinyin_answer and yindiao_entry.get() in yindiao_answer:
            self.passed = 'pass'
            return True

        elif pinyin1[-3:] == 'ing' or pinyin2[-3:] == 'ing':
            if pinyinentry[-2:] == 'in':
                
                if yindiao_entry.get() in yindiao_answer:
                    self.passed = 'pass'
                    return True

        elif pinyin1[-2:] == 'in' or pinyin2[-2:] == 'in':
            if pinyinentry[-3:] == 'ing':

                if yindiao_entry.get() in yindiao_answer:
                    self.passed = 'pass'
                    return True
        elif pinyin1[-2:] == 'en' or pinyin2[-2:] == 'en':
            if pinyinentry[-3:] == 'eng':
                if yindiao_entry.get() in yindiao_answer:
                    self.passed = 'pass'
                    return True

        elif pinyin1[-3:] == 'eng' or pinyin2[-3:] == 'eng':
            if pinyinentry[-2:] == 'en':
                
                if yindiao_entry.get() in yindiao_answer:
                    self.passed = 'pass'
                    return True

        else:
            self.passed = 'fail'
            return False

class ChineseAnswers:

    def __init__(self):
        self.character = ''
        self.pinyin = ''
        self.yindiao = 0


def read_dictionary():
    global list_of_answers
    doc = open(DictionaryFilePath, 'r', encoding='utf-8')
    tempstr = doc.read()
    a = tempstr.split('\n')
    list_of_answers = []
    for c in a:
        b = c.split(',')

        ca = ChineseAnswers()
        ca.character = b[0]
        ca.pinyin = b[1]
        ca.yindiao = int(b[2])
        list_of_answers.append(ca)
    return list_of_answers

def read_history():
    global list_of_objects
    doc = open(HistoryFilePath, 'r', encoding='utf-8')
    tempstr = doc.read()
    a = tempstr.split('\n')
    list_of_objects = []
    for c in a:
        b = c.split(',')

        #__init__ is used
        tr = TestRecord()
        tr.character = b[0]
        tr.pinyin = b[1]
        tr.yindiao = int(b[2])
        tr.passed = b[3]
        tr.time = datetime.datetime.strptime(b[4], '%c')
        list_of_objects.append(tr)
 
    return list_of_objects

def groupby(list):
    result = {}

    for current_object in list:
        key = current_object.character

        if key in result:
            temp_list = result.get(key)
            temp_list.append(current_object)
            result[key] = temp_list
        else:
            result[key] = [current_object]

    return result


def sort_time(dictionary):
    global words
    daysAgo_dictionary = {}
    CurrentTime = datetime.datetime.strptime(date.strftime('%c'), '%c')
    list_of_objects = []
    for key in dictionary:
        time = datetime.datetime.strptime('Fri Jan 1 0:00:01 2010', '%c')
        list_of_objects = dictionary.get(key)

        for j in range(0, len(list_of_objects)):          
            if list_of_objects[j].time > time:
                time = list_of_objects[j].time

        how_long_ago = CurrentTime - time
        how_many_days_ago = how_long_ago.days
        daysAgo_dictionary[key] = how_many_days_ago

    return daysAgo_dictionary

dictionary = groupby(read_history())
daysAgo_dictionary = sort_time(dictionary)

def sort_percentages(dictionaryOfAnswers, dictionary, daysAgo_dictionary):
    global words
    percentage_dictionary = {}
    list_of_objects = []
    for key in dictionary:
        number_of_trys = 0
        number_fail = 0
        number_pass = 0
        list_of_objects = dictionary.get(key)

        for j in range(0, len(list_of_objects)):
            if list_of_objects[j].passed == 'pass':
                number_pass = number_pass + 1
                number_of_trys = number_of_trys + 1
            else:
                number_fail = number_fail + 1
                number_of_trys = number_of_trys + 1

        percentage = float(  
            (number_pass - number_fail)-((daysAgo_dictionary[key]+1)/13)  
                          )
        percentage_dictionary[key] = percentage

    for key in dictionaryOfAnswers:
        if key not in percentage_dictionary:
            percentage_dictionary[key] = -100

    
    percentage_list = sorted(percentage_dictionary.items(), key=lambda x: x[1])
    # percentage_list.reverse()

    return percentage_list
    
a = read_dictionary()
dictionary_of_answers = groupby(read_dictionary())
dictionary = groupby(read_history())
percentage_list = sort_percentages(dictionary_of_answers,dictionary, daysAgo_dictionary)
def next_word():
    global Chinese_word_label4
    global count
    Chinese_Word_Variable2.set('')
    Chinese_Word_Variable3.set('')
    global dictionary_of_answers
    global dictionary
    global percentage_list
    global rank_of_word

    current_word = TestRecord()
    #entered word and pinyin and yindiaos
    current_word.character = percentage_list[rank_of_word][0]
    current_word.pinyin = pinyin_entry.get()
    current_word.yindiao = yindiao_entry.get()
    
    Chinese_Word_Variable2.set('')
    Chinese_Word_Variable3.set('')
    
    #coorect pinyin and yindiao
    coorect_word = dictionary_of_answers.get(current_word.character)
    Chinese_Word_Variable.set(current_word.character)
    pinyin_answer = []
    yindiao_answer = []
    for i in coorect_word:
        pinyin_answer.append(i.pinyin)
        yindiao_answer.append(i.yindiao)
    # pinyin_answer = coorect_word[0].pinyin
    # yindiao_answer = coorect_word[0].yindiao

    current_word.pinyin = pinyin_entry.get()
    current_word.yindiao = yindiao_entry.get()
    if current_word.character != '' and pinyin_entry.get() != "":
        count_Word_variable.set(str(count))
        count = count + 1

        if count > 50:
            Chinese_word_label4.config(fg="Green", font=('KaiTi', 200))
        

        if current_word.check_if_coorect(pinyin_answer, str(yindiao_answer)):
            current_word.time = date.strftime('%c')
            current_word.add_to_history(HistoryFilePath)
            rank_of_word = rank_of_word + 1
            current_word = TestRecord()
            Chinese_Word_Variable2.set('')
            Chinese_Word_Variable3.set('')
            current_word.character = percentage_list[rank_of_word][0]
            Chinese_Word_Variable.set(current_word.character)

        else:
            current_word.time = date.strftime('%c')
            current_word.add_to_history(HistoryFilePath)
            rank_of_word = rank_of_word + 1
            pinyin_answer = coorect_word[0].pinyin
            yindiao_answer = coorect_word[0].yindiao
            current_word.character = percentage_list[rank_of_word][0]
            Chinese_Word_Variable.set(current_word.character)
            Chinese_Word_Variable2.set(str(pinyin_answer))
            Chinese_Word_Variable3.set(str(yindiao_answer))

app = Tk()
app.title('Chinese Pratice')
app.geometry('1000x640')                                                        #, height=12, padx = 470
btn = Button(app, text='next word', command=next_word, font=('KaiTi', 40), padx = 550, height=3).grid(row=0, sticky=W)
Chinese_Word_Variable = StringVar()
Chinese_Word_Variable2 = StringVar()
Chinese_Word_Variable3 = StringVar()



count_Word_variable = StringVar()
global Chinese_word_label4
Chinese_word_label4 = Label(app, textvariable = count_Word_variable, padx = 20, font=('Noto Sans SC', 20))
Chinese_word_label4.grid(row=6)
Chinese_Word_Label = Label(app, textvariable=Chinese_Word_Variable, padx = 20, font=('KaiTi', 40)).grid(row=1, column=0)
Chinese_Word_Label2 = Label(app, textvariable=Chinese_Word_Variable2, padx = 20, font=('Noto Sans SC', 15)).grid(row=2)
Chinese_Word_Label3 = Label(app, textvariable=Chinese_Word_Variable3, padx = 20, font=('Noto Sans SC', 15)).grid(row=3)


# Chinese_Word_Label.config(width=200, fontsize=50)
#Chinese_Word_Variable.set(current_word.character)

#make entrys for pinyin and yindiao
pinyin_label = Label(app, text='enter pinyin', font=('Noto Sans SC', 20)).grid(row=4, sticky=W)
yindiao_label = Label(app, text='enter yindiao', font=('Noto Sans SC', 20)).grid(row=5, sticky=W)
pinyin_entry = Entry(app, font=('Noto Sans SC', 20))
pinyin_entry.grid(row=4)
yindiao_entry = Entry(app,font=('Noto Sans SC', 20) )
yindiao_entry.grid(row=5)
entered_yindiao = 0
entered_pinyin = 0

app.mainloop()
